### PR TITLE
Add autotest for devo telemetry 

### PR DESCRIPTION
**Pull request by shiv-tyagi**
This adds an autotest for devo telemetry (issue #19174).
This also fixes the method `gpsDdToDmsFormat` in devo telemetry driver which returned 0 for valid values which lead `devoPacket` packet sending `lat` and `lon` as 0 always.

### CODE REVIEW

File: libraries/AP_Devo_Telem/AP_Devo_Telem.cpp [hunk 1]
```diff
@@ -48,12 +48,12 @@ void AP_DEVO_Telem::init()
 }
 
 
-uint32_t AP_DEVO_Telem::gpsDdToDmsFormat(float ddm)
+uint32_t AP_DEVO_Telem::gpsDdToDmsFormat(int32_t ddm)
 {
-    int32_t deg = (int32_t)ddm;
-    float mm = (ddm - deg) * 60.0f;
+    int32_t deg = (int32_t)(ddm / 1e7);
+    double mm = (ddm / 1.0e7 - deg) * 60.0f;
```
Comment (peterbarker) [inline libraries/AP_Devo_Telem/AP_Devo_Telem.cpp:54]:
```suggestion
    int32_t deg = (int32_t)(ddm * 1e-7);
    float mm = (ddm * 1.0e-7 - deg) * 60.0f;
```

Multiplications are *much* cheaper than divisions.  That's important in the embedded sphere.

File: Tools/autotest/common.py [hunk 2]
```diff
@@ -10771,6 +10847,82 @@ def test_ltm(self):
...
 
+    def convertDdToDmsFormat(self, ddm):
+        '''converts decimal degree values to degree minutes'''
+        deg = math.trunc(ddm / 1e7)
+        mm = (deg * 100 + ((ddm / 1e7) - deg) * 60) / 100
+        if mm < -180.0 or mm > 180.0:
+            mm = 0.0
+        return math.trunc(mm * 1.0e7)
+
+    def test_devo(self):
```
Comment (peterbarker) [inline Tools/autotest/common.py:10858]:
```suggestion
    def DEVO(self):
```

Trying to be more consistent about making these methods the same name as the test.

File: libraries/AP_Devo_Telem/AP_Devo_Telem.cpp [hunk 3]
```diff
@@ -48,12 +48,12 @@ void AP_DEVO_Telem::init()
...
 
-    mm = ((float)deg * 100.0f + mm) /100.0f;
+    mm = ((double)deg * 100.0f + mm) /100.0f;
```
Comment (peterbarker) [inline libraries/AP_Devo_Telem/AP_Devo_Telem.cpp:56]:
```suggestion
    mm = ((float)deg * 100.0f + mm) /100.0f;
```

The embedded hardware we run on doesn't necessarily have the hardware to run double-precision maths.  That puts a huge burden on the CPU as it has to emulate it in software.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Not much wrong here!

File: libraries/AP_Devo_Telem/AP_Devo_Telem.cpp (hunk 1 above)
Comment (shiv-tyagi) [reply libraries/AP_Devo_Telem/AP_Devo_Telem.cpp]: Done!

File: Tools/autotest/common.py (hunk 2 above)
Comment (shiv-tyagi) [reply Tools/autotest/common.py]: Done!

File: libraries/AP_Devo_Telem/AP_Devo_Telem.cpp (hunk 3 above)
Comment (shiv-tyagi) [reply libraries/AP_Devo_Telem/AP_Devo_Telem.cpp]: Done!

**Comment by shiv-tyagi** [comment]
@peterbarker Thanks for reviewing it. I have done the requested changes. Can we merge it?

**Review by peterbarker — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**